### PR TITLE
Let GitHub generates the release notes

### PR DESCRIPTION
**Pull request by bmarty**
## Type of change

- [ ] Feature
- [ ] Bugfix
- [ ] Technical
- [x] Other : release note

## Content

Remove the usage of towncrier and let GitHub generate the release note

Labels have been created: https://github.com/element-hq/element-x-android/labels?q=PR-

## Motivation and context

No need to create towncrier file anymore and rely on GitHub

Use same DevX than on iOS. Inspired by https://github.com/element-hq/element-x-ios/pull/2966

Limitation: during a release the file CHANGES.md will have to be updated after the release has been generated. We keep this file for an easy way to grep releases content.

## Tests

- Nothing particular, if CI is happy with this PR this is fine
- Need to be carefull for the next release, ideally the folder `./changelog.d ` will have to be deleted after the next release.

## Tested devices

NA

## Checklist

- [ ] Changes have been tested on an Android device or Android emulator with API 23
- [ ] UI change has been tested on both light and dark themes
- [ ] Accessibility has been taken into account. See https://github.com/element-hq/element-x-android/blob/develop/CONTRIBUTING.md#accessibility
- [ ] Pull request is based on the develop branch
- [ ] Pull request includes a new file under ./changelog.d. See https://github.com/element-hq/element-x-android/blob/develop/CONTRIBUTING.md#changelog
- [ ] Pull request includes screenshots or videos if containing UI changes
- [ ] Pull request includes a [sign off](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#sign-off)
- [ ] You've made a self review of your PR

### CODE REVIEW

**Comment by bmarty** [comment]
image

Good!

File: .github/workflows/fork-pr-notice.yml [hunk 1]
```diff
@@ -25,7 +25,7 @@ jobs:
               body: `Thank you for your contribution! Here are a few things to check in the PR to ensure it's reviewed as quickly as possible:
             
               - Your branch should be based on \`origin/develop\`, at least when it was created.
-              - There is a changelog entry in the \`changelog.d\` folder with [the Towncrier format](https://towncrier.readthedocs.io/en/latest/tutorial.html#creating-news-fragments).
+              - The title of the PR will be used for release notes, so it need to describe the change visible to the user.
```
Comment (jmartinesp) [inline .github/workflows/fork-pr-notice.yml:28]: Nit: `it needs`

Comment (bmarty) [reply .github/workflows/fork-pr-notice.yml]: https://github.com/element-hq/element-x-android/pull/3105/commits/790c0aa0c0dc8399e8f6691f3b2fe725097a99b4

File: .github/pull_request_template.md [hunk 2]
```diff
@@ -55,7 +55,7 @@ Uncomment this markdown table below and edit the last line `|||`:
 - [ ] UI change has been tested on both light and dark themes
 - [ ] Accessibility has been taken into account. See https://github.com/element-hq/element-x-android/blob/develop/CONTRIBUTING.md#accessibility
 - [ ] Pull request is based on the develop branch
-- [ ] Pull request includes a new file under ./changelog.d. See https://github.com/element-hq/element-x-android/blob/develop/CONTRIBUTING.md#changelog
+- [ ] Pull request title will be used in the release note, it clearly define what will change for the user
```
Comment (jmartinesp) [inline .github/pull_request_template.md:58]: Should we mention the usage of label heres too?

**Review by jmartinesp — APPROVED** [review]
I haven't tested the release script, but the changes seem fine to me. Thanks!

File: .github/pull_request_template.md (hunk 2 above)
Comment (bmarty) [reply .github/pull_request_template.md:58]: Danger will complain if the label is missing, so I do not think it worth it. Also I do not think external contributors using forks can add labels (not sure about this).

**Pull request merged**